File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        x = sum = 0
        d = 1
        while(n):
            r = n % 10
            if(r != 0):
                x = r * d + x
                d *= 10
                sum += r
            n //= 10
        return sum * x
        

File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.cpp
class Solution {
public:
    long long sumAndMultiply(int n) {
        long long sum = 0;
        string x = "";
        while(n > 0){
            int r = n % 10;
            if(r != 0){
                x = to_string(r) + x;
                sum += r;
            }
            n /= 10;
        }
        return x != "" ? stoi(x) * sum : 0;
    }
};